openedx/features/wikimedia_features/messenger/api/v0/views.py

```diff
@@ -0,0 +1,189 @@
+            {
+                "id": 37,
+                "with_user": "staff",
+                "last_message": "hello, how are you d...",
+                "unread_count": 0,
+                "with_user_img": "profile_image_url"
+            },
+           ...
+        ],
+    }
+
+    Retrieve single inbox message object
+    Get /messenger/api/v0/inbox/pk
+    {
+        "id": 37,
+        "with_user": "staff",
+        "last_message": "hello, how are you d...",
+        "unread_count": 0,
+        "with_user_img": "profile_image_url"
+    }
+    ```
+
+    Update inbox message object
+    PATCH /messenger/api/v0/inbox/pk
+    {
+        "unread_count": 2,
+    }
+    ```
+
+    Note:
+    - last_message will only contain first 20 chars of full message.
+    - user can view and update only his inbox resources
+
+    """
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = InboxSerializer
+    pagination_class = MesssengerResultsSetPagination
+
+    def get_queryset(self):
+        return Inbox.user_inbox.find_all(self.request.user)
+
+
+class ConversationView(viewsets.ModelViewSet):
+    """
+    Return conversation between two users -> All messages sent or received between two users.
+    User can only view his related conversations
+    GET /conversation/?user1=username1&user2=username2
+    [
+        {
+            "id": 2093,
+            "sender": "honor",
+            "receiver": "edx",
+            "message": "what's the progress on task1 ?",
+            "created": "08/31/21-01:09 PM",
+            "sender_img": "sender_profile_image url"
+        },
+        ...
+    ]
+    """
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, )
+    serializer_class = ConversationSerializer
+    pagination_class = MesssengerResultsSetPagination
+
+    def get_queryset(self):
+        serializer = ConversationAccessSerializer(data=self.request.GET)
+        if serializer.is_valid(raise_exception=True):
+            user1 = serializer.validated_data.get('user1')
+            user2 = serializer.validated_data.get('user2')
+            if self.request.user in [user1, user2]:
+                return Message.chat.history(user1, user2)
+            raise PermissionDenied("You can not access other user's chat")
+
+
+class MessageCreateView(generics.CreateAPIView):
+    """
+    Create Single Message -
+    POST /messenger/api/v0/message/
+    {
+        "sender": "sender_username"
+        "receiver": "receiver_username",
+        "message": "sample message text"
+    }
+    Return newly created message
+    {
+        "id": 2092,
+        "sender": "sender_username",
+        "receiver": "receiver_username",
+        "message": "sample message text",
+        "created": "08/31/21-01:08 PM",
+        "sender_img": "sender_profile_image_url"
+    }
+
+    """
+    authentication_classes = (SessionAuthentication,)
```

Comment:
we aren't consistent with this authentication class, this isn't being used in Bulk creation.